falcon/api_helpers.py

```diff
@@ -121,9 +116,29 @@ def default_serialize_error(req, exception):
             preferred = 'application/xml'
 
     if preferred is not None:
+        resp.append_header('Vary', 'Accept')
```

Comment:
Have added an assertIn to the `test_base_class`. And dealt with all the other comments; have done it as a fixup commit so it’s easy to compare, let me know if you’d like me to rebase/squash it first.